### PR TITLE
Fix all Java warnings on the project

### PR DESCRIPTION
**Pull request by spannm**
This pull request fixes all Java warnings such as redundant imports, unneeded SuppressWarnings annotations, resource leaks

Review was done in Eclipse IDE with following Java 8 JDK:
```
java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)
```

### CODE REVIEW

File: src/main/java/com/healthmarketscience/jackcess/util/MemFileChannel.java [hunk 1]
```diff
@@ -116,13 +116,9 @@ public static MemFileChannel newChannel(File file) throws IOException {
...
     throws IOException
   {
-    FileChannel in = null;
-    try {
-      return newChannel(in = new RandomAccessFile(
-                            file, RO_CHANNEL_MODE).getChannel(),
-                        mode);
-    } finally {
-      ByteUtil.closeQuietly(in);
+    try (RandomAccessFile raf = new RandomAccessFile(file, RO_CHANNEL_MODE);
```
Comment (jahlborn) [inline src/main/java/com/healthmarketscience/jackcess/util/MemFileChannel.java:119]: the original code was intentional.  i didn't want errors when closing the file channel to fail the method.

File: src/main/java/com/healthmarketscience/jackcess/impl/IndexData.java [hunk 2]
```diff
@@ -532,6 +532,7 @@ protected static void writeDefinitions(
    * @param creator description of the indexes to write
    * @param buffer Buffer to write to
    */
+  @SuppressWarnings("resource")
```
[the remark this answers is not in the document]
Comment (spannm) [reply src/main/java/com/healthmarketscience/jackcess/impl/IndexData.java:535]: The `DatabaseImpl` returned by `creator.getDatabase()`. However `creator` is a method parameter, so the method should not have responsibility to close the database. This warning seems a false positive. Therefore I annotated with `SuppressWarnings`.
You're not getting a warning in line 564 `throw new IllegalArgumentException(...`?

File: src/main/java/com/healthmarketscience/jackcess/util/MemFileChannel.java (hunk 1 above)
Comment (spannm) [reply src/main/java/com/healthmarketscience/jackcess/util/MemFileChannel.java:119]: I see.
Updated this part as follows. This is functionally equivalent but removes the Java warning:
```
{
  RandomAccessFile raf = null;
  FileChannel in = null;
  try {
    raf = new RandomAccessFile(file, RO_CHANNEL_MODE);
    return newChannel(in = raf.getChannel(), mode);
  } finally {
    ByteUtil.closeQuietly(in);
    ByteUtil.closeQuietly(raf);
  }
}

```

File: src/main/java/com/healthmarketscience/jackcess/impl/IndexData.java [hunk 3]
```diff
@@ -1604,13 +1605,13 @@ private static int calcMaxPageEntrySize(JetFormat format)
...
                             getName());
   }
 
-  private static String withErrorContext(String msg, DatabaseImpl db,
+  private static String withErrorContext(String msg, String dbName,
                                          String tableName, String idxName) {
-    return msg + " (Db=" + db.getName() + ";Table=" + tableName +
+    return msg + " (Db=" + dbName + ";Table=" + tableName +
       ";Index=" + idxName + ")";
   }
```
[the remark this answers is not in the document]
Comment (spannm) [reply src/main/java/com/healthmarketscience/jackcess/impl/IndexData.java:1616]: > this forces all the callers [..]

Parameter changed while chasing resource leakage of Database object in class `IndexData`. There are only two callers.

[the remark this answers is not in the document]
Comment (jahlborn) [reply src/main/java/com/healthmarketscience/jackcess/impl/IndexData.java:1616]: right, but it makes the method less convenient.  i like clarifying java warnings, but i don't love making code less ergonomic for no real benefit (other than silencing a superfluous warning).

**Comment by spannm** [comment]
No problem @jahlborn. Original method signature restored and PR updated.

File: src/main/java/com/healthmarketscience/jackcess/impl/RowImpl.java [hunk 4]
```diff
@@ -101,7 +101,6 @@ public Double getDouble(String name) {
   }
 
   @Override
-  @SuppressWarnings("deprecation")
```
Comment (jahlborn) [inline src/main/java/com/healthmarketscience/jackcess/impl/RowImpl.java:104]: So i tried to apply these changes last night and i could no longer compile the project with these changes.  I'm not sure what version of java you are using, but i believe all of these deprecation suppressions are still needed (the project is compiled with java 8).

**Comment by jahlborn** [comment]
i incorporated the changes that i could.  i kept the suppression for deprecation needed to get the project to compile.

**Comment by spannm** [comment]
I am glad you incorporated most of the changes, @jahlborn! Thanks

Some local var/method unused" warnings remain in `IndexCodesTest` (due to commented out code)

> i kept the suppression for deprecation needed to get the project to compile.

What do you mean by that?

**Comment by jahlborn** [comment]
> I am glad you incorporated most of the changes, @jahlborn! Thanks
> 
> Some local var/method unused" warnings remain in `IndexCodesTest` (due to commented out code)

yeah, that is useful code, i didn't want to litter it with "unused" suppressions.  it's only test code.

> > i kept the suppression for deprecation needed to get the project to compile.
> 
> What do you mean by that?

If i remove the deprecation suppression from RowImpl, this is what i get when i compile:

~~~~~
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ jackcess ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 192 source files to /home/james/src/jackcess/target/classes
[WARNING] /home/james/src/jackcess/src/main/java/com/healthmarketscience/jackcess/impl/RowImpl.java:[105,15] getDate(java.lang.String) in com.healthmarketscience.jackcess.Row has been deprecated
[INFO]
~~~~~

**Comment by spannm** [comment]
> If i remove the deprecation suppression from RowImpl, this is what i get when i compile:

I can confirm.

I mostly use Eclipse IDE, and Eclipse incorrectly claims these suppressions should be removed, which must be due to [bug 536256](https://bugs.eclipse.org/bugs/show_bug.cgi?id=536256), reported in 2018 but never fixed :(

Shall we close this PR?

**Comment by jahlborn** [comment]
> > If i remove the deprecation suppression from RowImpl, this is what i get when i compile:
> 
> I can confirm.
> 
> I mostly use Eclipse IDE, and Eclipse incorrectly claims these suppressions should be removed, which must be due to [bug 536256](https://bugs.eclipse.org/bugs/show_bug.cgi?id=536256), reported in 2018 but never fixed :(
> 
> Shall we close this PR?

yep, thanks for your help!

**Pull request closed without merging**